### PR TITLE
Feature/login page

### LINKED ISSUE #244: Create Authentication Page
References user story #166

As a user I should have a place that I can sign into if I'm not authenticated. I should be able to enter in a username and a password. Once entered I should be able to submit them. If username or password fields are empty, I should not be able to submit and an error should be displayed.

### PR DESCRIPTION
**Pull request by Kenny-Miller**
Closes #244

image

Creates a basic login page that takes in a username, password, and remember me checkbox. All input fields have validation. Errors will only appear if the input fields are invalid and the user has tried submitting the form at least once. Fields are valid if the input is in-between 8-24 characters.

### CODE REVIEW

File: BEIMA.Client/src/pages/Authentication/LoginPage.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  useEffect(() => {
+    setPageName('Login')
+  }, [setPageName])
+
+
+  const ValidateUsername = (username) => {
+    let valid = true
+    if(username.length < 8 || username.length > 24){
+      valid = false
+    }
```
Comment (Kenny-Miller) [inline BEIMA.Client/src/pages/Authentication/LoginPage.js:26]: Improvements are welcome for the validation. These are just here as a place holder as we haven't discussed what should be allowed.

File: BEIMA.Client/src/pages/Authentication/LoginPage.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  const [submitting, setSubmitting] = useState(false)
+
+  useEffect(() => {
+    setPageName('Login')
+  }, [setPageName])
+
+
+  const ValidateUsername = (username) => {
+    let valid = true
+    if(username.length < 8 || username.length > 24){
```
Comment (josephmoore99) [inline BEIMA.Client/src/pages/Authentication/LoginPage.js:24]: I could see username being more than 24 characters, especially if we use emails.

File: BEIMA.Client/src/pages/Authentication/LoginPage.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      valid = false
+    }
+    return valid
+  }
+
+  const ValidatePassword = (password) => {
+    let valid = true
+    if(password.length < 8 || password.length > 24){
```
Comment (josephmoore99) [inline BEIMA.Client/src/pages/Authentication/LoginPage.js:32]: I don't think we should have a cap on password length, or I think we should make it ridiculously big like 100 characters or something.

File: BEIMA.Client/src/App.js [hunk 4]
```diff
@@ -25,6 +26,7 @@ const App = () => {
       <Routes>
         <Route path="/" element={<AppLayout />}>
           <Route index element={<HomePage />} />
+          <Route path="login" element={<LoginPage/>}/>
```
Comment (AshlynAdamson) [inline BEIMA.Client/src/App.js:29]: We should make the login page the landing page

Comment (josephmoore99) [reply BEIMA.Client/src/App.js:29]: I that will likely depend on their login session or whatnot, so I would suggest waiting to do that until we actually have logins working.

**Comment by roguedeleter** [comment]
Should we still be showing the sidebar if the user is not logged in?

File: BEIMA.Client/src/pages/Help/HelpPage.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { Card } from "react-bootstrap";
 import { useOutletContext } from 'react-router-dom';
 import { useEffect } from 'react';
-import './HelpPage.css'
+import styles from'./HelpPage.module.css'
```
Comment (roguedeleter) [inline BEIMA.Client/src/pages/Help/HelpPage.js:4]: Does this still compile properly without a space between "from" and the file path?

**Comment by Kenny-Miller** [comment]
> Should we still be showing the sidebar if the user is not logged in?

Probably not. That will have to be a future task when we implement auth

File: BEIMA.Client/src/pages/Help/HelpPage.js (hunk 5 above)
Comment (AshlynAdamson) [reply BEIMA.Client/src/pages/Help/HelpPage.js:4]: I think the build.sh would have gotten angry if it didn't, so we should be good

**Review by roguedeleter — DISMISSED** [review]

**Review by AshlynAdamson — APPROVED** [review]

File: BEIMA.Client/src/pages/Help/HelpPage.js (hunk 5 above)
Comment (Kenny-Miller) [reply BEIMA.Client/src/pages/Help/HelpPage.js:4]: I'll update it regardless

**Review by josephmoore99 — APPROVED** [review]

**Review by roguedeleter — APPROVED** [review]

**Review by AshlynAdamson — APPROVED** [review]

**Pull request merged**